packages/lesswrong/themes/userThemes/darkMode.ts

```diff
@@ -44,6 +49,60 @@ export const invertedGreyscale = {
 
 const greyAlpha = (alpha: number) => `rgba(255,255,255,${alpha})`;
 
+// CkEditor allows users to provide colors for table cell backgrounds and
+// borders, which get embedded into the HTML looking like this:
+//   <td style="background-color: rgb(1,2,3)">
+// User-provided colors can be in any format that CSS accepts, and may set
+// the `background-color` and `border` properties on <table> and <td>
+// elements.
+//
+// We don't have any theme-specific HTML postprocessing, so this poses a bit of
+// a problem. Our solution is to override the `background-color` and `border`
+// properties on <td> and <table> to a safe default, to guarantee there won't
+// be any black-on-black or white-on-white text, and then then use CSS
+// attribute matching to replace specific colors that have been used in
+// posts we care about with proper aesthetic replacements. The CSS winds up
+// looking like this:
+//   .content td, .content table {
+//     background-color: black !important;
+//     border-color: #333 !important;
+//   }
+//   .content td[style*="background-color:rgb(230, 230, 230)"], .content table[style*="background-color:rgb(230, 230, 230)"] {
+//     background-color: #202020 !important;
+//   }
+// (Not real color values, but real syntax.)
+//
+const safeColorFallbacks = `
+.content td[style*="background-color:"], .content table[style*="background-color:"] {
+  background-color: black !important;
+  border-color: #333 !important;
```

Comment:
Unfortunately `filter: invert(1)` doesn't work. The issue is that we get customized background colors (because CkEditor has UI that allows for changing the table color), but those elements don't have a corresponding customized text color, they just inherit the text color from the rest of the page. So in normal mode you have a custom background color, let's say light-grey, and an inherited-default text color, which is black. In dark mode, you have the same light-grey background color, but the text is now white, so you have white-on-white which is unreadable. If the `filter` property changed only the background that would be great, but unfortunately it changes both the background and the text colors, so it winds up changing it from white-on-white to black-on-black, which isn't any better.

Reply:
So then
```
.content td[style*="background-color:"] {
   color: #000,
   filter: invert(1)
}
```
?

Could you maybe send me a link to a post that has this problem? It could be that I'm just misunderstanding what the issue is. But I feel like adding ~150 lines to support a couple legacy posts is a lot of technical debt, and seems worth thinking of a way that we could have fewer special cases.

Reply:
Here's an example of a post where this is important: https://www.lesswrong.com/s/n945eovrA3oDueqtq/p/tcCxPLBrEXdxN5HCQ . Your proposed solution does not successfully prevent black-on-black text, because applying a `color` to the `td` doesn't reliably change the color of all the descendent elements.

Reply:
(Note: If you look at that link on LW now it looks fine because I already merged this to there.)